catalyst/contrib/dl/callbacks/wandb_logger.py

```diff
@@ -177,13 +178,22 @@ def on_epoch_end(self, runner: IRunner):
         )
 
         if self.log_on_epoch_end:
-            self._log_metrics(
-                metrics=splitted_epoch_metrics[extra_mode],
-                step=runner.global_epoch,
-                mode=extra_mode,
-                suffix=self.epoch_log_suffix,
-                commit=True,
-            )
+            try:
+                self._log_metrics(
+                    metrics=splitted_epoch_metrics[extra_mode],
+                    step=runner.global_epoch,
+                    mode=extra_mode,
+                    suffix=self.epoch_log_suffix,
+                    commit=True,
+                )
+            except KeyError:
+                self._log_metrics(
+                    metrics=splitted_epoch_metrics,
```

Comment:
there is the difference between params, `splitted_epoch_metrics[extra_mode]` vs `splitted_epoch_metrics`. why so?

Reply:
I attached issue to this PR. Sometimes I get an KeyError both in config API and in notebook API. Here is an example. Just try to run it
```python
import os
import torch
from torch.nn import functional as F
from torch.utils.data import DataLoader
from catalyst import dl
from catalyst.contrib.data.transforms import ToTensor
from catalyst.contrib.datasets import MNIST
from catalyst.utils import metrics
from catalyst.contrib.dl import WandbLogger

model = torch.nn.Linear(28 * 28, 10)
optimizer = torch.optim.Adam(model.parameters(), lr=0.02)

loaders = {
    "train": DataLoader(MNIST(os.getcwd(), train=True, download=True, transform=ToTensor()), batch_size=32),
    "valid": DataLoader(MNIST(os.getcwd(), train=False, download=True, transform=ToTensor()), batch_size=32),
}

class CustomRunner(dl.Runner):

    def predict_batch(self, batch):
        # model inference step
        return self.model(batch[0].to(self.device).view(batch[0].size(0), -1))

    def _handle_batch(self, batch):
        # model train/valid step
        x, y = batch
        y_hat = self.model(x.view(x.size(0), -1))

        loss = F.cross_entropy(y_hat, y)
        accuracy01, accuracy03 = metrics.accuracy(y_hat, y, topk=(1, 3))
        self.batch_metrics.update(
            {"loss": loss, "accuracy01": accuracy01, "accuracy03": accuracy03}
        )

        if self.is_train_loader:
            loss.backward()
            self.optimizer.step()
            self.optimizer.zero_grad()

runner = CustomRunner()
# model training
runner.train(
    model=model,
    optimizer=optimizer,
    loaders=loaders,
    logdir="./logs",
    num_epochs=5,
    verbose=True,
    load_best_on_end=True,
    callbacks={
        "logger": WandbLogger(
        )
    },
    timeit=True
)
```